code/modules/projectiles/guns/energy.dm

```diff
@@ -594,3 +594,776 @@
+	fire_delay = 10
+	fire_sound = 'sound/weapons/guns/fire/disabler.ogg'
+	message_to_user = "You set the Laser Pistol's charge mode to disabler fire."
+	fire_mode = GUN_FIREMODE_AUTOMATIC
+	icon_state = "tep"
+	radial_icon_state = "laser_disabler"
+
+/datum/energy_pistol_mode/heat
+	charge_cost = 110
+	ammo = /datum/ammo/energy/lasgun/marine/pistol/heat
+	fire_delay = 0.5 SECONDS
+	fire_sound = 'sound/weapons/guns/fire/laser3.ogg'
+	message_to_user = "You set the Laser Pistol's charge mode to wave heat."
+	fire_mode = GUN_FIREMODE_AUTOMATIC
+	icon_state = "tep"
+	radial_icon_state = "laser_heat"
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine_pistol/unique_action(mob/user)
+	return switch_modes(user)
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine_pistol/proc/switch_modes(mob/user)
+	if(!user)
+		CRASH("switch_modes called with no user.")
+
+	var/list/available_modes = list()
+	for(var/mode in mode_list)
+		available_modes += list("[mode]" = image(icon = initial(mode_list[mode].radial_icon), icon_state = initial(mode_list[mode].radial_icon_state)))
+
+	var/datum/energy_pistol_mode/choice = mode_list[show_radial_menu(user, user, available_modes, null, 64, tooltips = TRUE)]
+	if(!choice)
+		return
+
+
+	playsound(user, 'sound/weapons/emitter.ogg', 5, FALSE, 2)
+
+
+	gun_firemode = initial(choice.fire_mode)
+
+
+	charge_cost = initial(choice.charge_cost)
+	ammo = GLOB.ammo_list[initial(choice.ammo)]
+	fire_delay = initial(choice.fire_delay)
+	fire_sound = initial(choice.fire_sound)
+	SEND_SIGNAL(src, COMSIG_GUN_AUTOFIREDELAY_MODIFIED, fire_delay)
+	SEND_SIGNAL(src, COMSIG_GUN_FIRE_MODE_TOGGLE, initial(choice.fire_mode), user.client)
+
+	base_gun_icon = initial(choice.icon_state)
+	update_icon()
+
+	to_chat(user, initial(choice.message_to_user))
+
+	var/obj/screen/ammo/A = user.hud_used.ammo //The ammo HUD
+	A.update_hud(user)
+
+	return TRUE
+
+
+
+//Ammo/Charge functions
+/obj/item/weapon/gun/energy/lasgun/standard_marine_pistol/update_icon(mob/user)
+	var/cell_charge = (!cell || cell.charge <= 0) ? 0 : CEILING((cell.charge / max(cell.maxcharge, 1)) * 100, 25)
+	icon_state = "[base_gun_icon]_[cell_charge]"
+	update_mag_overlay(user)
+
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine_pistol/update_item_state(mob/user)// lasgun code thing i don't want so redfined
+    . = item_state
+    item_state = "[initial(icon_state)][flags_item & WIELDED ? "_w" : ""]"
+    if(. != item_state && ishuman(user))
+        var/mob/living/carbon/human/human_user = user
+        if(src == human_user.l_hand)
+            human_user.update_inv_l_hand()
+        else if (src == human_user.r_hand)
+            human_user.update_inv_r_hand()
+
+
+//TE Standard Laser Carbine
+
+obj/item/weapon/gun/energy/lasgun/standard_marine_carbine
+	name = "\improper TE-C Laser carbine"
+	desc = "A TerraGov standard issue laser carbine with an integrated charge selector for normal and scatter settings. Uses standard TE cells."
```

Comment:
```suggestion
	desc = "A TerraGov standard issue laser carbine with an integrated charge selector for normal and scatter settings. This variant is a carbine, commonly known as an energy repeater not only because of its automatic fire mode, but it is easier to handle in a pinch. Uses standard TE cells."
```

Reply:
not naming it a repeater.